### PR TITLE
Adding service module

### PR DESCRIPTION
**Pull request by devagupt**
- Service module in salt

### CODE REVIEW

File: hubblestack/utils/systemd.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+
+def booted(context=None):
+    '''
+    Return True if the system was booted with systemd, False otherwise.  If the
+    loader context dict ``__context__`` is passed, this function will set the
+    ``salt.utils.systemd.booted`` key to represent if systemd is running and
+    keep the logic below from needing to be run again during the same salt run.
+    '''
+    contextkey = 'salt.utils.systemd.booted'
```
Comment (goravsingal) [inline hubblestack/utils/systemd.py]: it has to be "hubblestack.utils.systemd.booted". 
I added hubblestack.utils.systemd. So you don't have to add systemd.py file from salt..

File: hubblestack/utils/systemd.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    if _sd_version is None:
+        return False
+    return _sd_version >= 205
+
+def version(context=None):
+    '''
+    Attempts to run systemctl --version. Returns None if unable to determine
+    version.
+    '''
+    contextkey = 'salt.utils.systemd.version'
```
Comment (goravsingal) [inline hubblestack/utils/systemd.py]: change from salt to hubblestack

File: tests/unittests/modules/test_systemd_service.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                )
+
+    def test_available(self):
+        '''
+        Test to check that the given service is available
+        '''
+        mock = MagicMock(side_effect=lambda x: _SYSTEMCTL_STATUS[x])
+
+        # systemd < 231
+        with patch.dict(systemd.__context__, {'salt.utils.systemd.version': 230}):
```
Comment (goravsingal) [inline tests/unittests/modules/test_systemd_service.py]: Same here, salt to hubblestack at all places in this file.

File: tests/unittests/utils/test_systemd.py [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+        # Ensure that os.stat returns True. os.stat doesn't return a bool
+        # normally, but the code is doing a simple truth check on the return
+        # data, so it is sufficient enough to mock it as True for these tests.
+        with patch('os.stat', side_effect=_booted_effect):
+            # Test without context dict passed
+            self.assertTrue(_systemd.booted())
+            # Test that context key is set when context dict is passed
+            context = {}
+            self.assertTrue(_systemd.booted(context))
+            self.assertEqual(context, {'salt.utils.systemd.booted': True})
```
Comment (goravsingal) [inline tests/unittests/utils/test_systemd.py]: Change salt to hubblestack

File: tests/unittests/utils/test_systemd.py [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+        '''
+        # Ensure that os.stat raises an exception even if test is being run on
+        # a systemd-booted host.
+        with patch('os.stat', side_effect=_not_booted_effect):
+            # Test without context dict passed
+            self.assertFalse(_systemd.booted())
+            # Test that context key is set when context dict is passed
+            context = {}
+            self.assertFalse(_systemd.booted(context))
+            self.assertEqual(context, {'salt.utils.systemd.booted': False})
```
Comment (goravsingal) [inline tests/unittests/utils/test_systemd.py]: Change salt to hubblestack
Change at all places in this file

**Review by goravsingal — CHANGES_REQUESTED** [review]

File: hubblestack/utils/systemd.py (hunk 1 above)
Comment (devagupt) [reply hubblestack/utils/systemd.py]: fixed

File: hubblestack/utils/systemd.py (hunk 2 above)
Comment (devagupt) [reply hubblestack/utils/systemd.py]: done

File: tests/unittests/modules/test_systemd_service.py (hunk 3 above)
Comment (devagupt) [reply tests/unittests/modules/test_systemd_service.py]: fixed

File: tests/unittests/utils/test_systemd.py (hunk 4 above)
Comment (devagupt) [reply tests/unittests/utils/test_systemd.py]: fixed

File: tests/unittests/utils/test_systemd.py (hunk 5 above)
Comment (devagupt) [reply tests/unittests/utils/test_systemd.py]: fixed

**Pull request merged**